qemu/tests/win_nics_teaming.py

```diff
@@ -0,0 +1,115 @@
+    nics = params.objects("nics")
+    ifnames = ()
+    for i in range(len(nics)):
+        mac = vm.get_mac_address(i)
+        connection_id = utils_net.get_windows_nic_attribute(session_serial,
+                                                            "macaddress",
+                                                            mac,
+                                                            "netconnectionid")
+        ifnames += (connection_id,)
+
+    # get params of teaming
+    setup_cmd = params["setup_cmd"]
+    status, output = session_serial.cmd_status_output(setup_cmd % ifnames)
+    if status:
+        test.fail("Failed to setup team nic from powershell,"
+                  "status=%s, output=%s" % (status, output))
+
+    # prepare test data
+    guest_path = (tmp_dir + "src-%s" % utils_misc.generate_random_string(8))
+    host_path = os.path.join(test.tmpdir, "tmp-%s" %
+                             utils_misc.generate_random_string(8))
+    logging.info("Test setup: Creating %dMB file on host", filesize)
+    process.run(dd_cmd % host_path, shell=True)
+
+    try:
+        netsh_set_cmd = "netsh interface set interface \"%s\" %s"
+        # transfer data
+        original_md5 = crypto.hash_file(host_path, algorithm="md5")
+        logging.info("md5 value of data original: %s" % original_md5)
+        logging.info("Failover test with file transfer")
+        transfer_thread = utils_misc.InterruptedThread(
+            vm.copy_files_to, (host_path, guest_path))
+        transfer_thread.start()
+        try:
+            while transfer_thread.isAlive():
+                for ifname in ifnames:
+                    session_serial.cmd(netsh_set_cmd % (ifname, "disable"))
+                    time.sleep(random.randint(1, 30))
+                    session_serial.cmd(netsh_set_cmd % (ifname, "enable"))
+                    time.sleep(random.randint(1, 30))
+        except aexpect.ShellProcessTerminatedError:
+            transfer_thread.join(suppress_exception=True)
+            raise
+        else:
+            transfer_thread.join()
+
+        os.remove(host_path)
+        logging.info('Cleaning temp file on host')
+        logging.info("Failover test 2 with file transfer")
+        transfer_thread = utils_misc.InterruptedThread(
+            vm.copy_files_from, (guest_path, host_path))
+        transfer_thread.start()
+        try:
+            nic_num = len(ifnames)
+            index = 0
+            while transfer_thread.isAlive():
+                index = index % nic_num
+                for num in range(nic_num):
+                    if num == index:
+                        session_serial.cmd(netsh_set_cmd %
+                                           (ifnames[num], "enable"))
+                    else:
+                        session_serial.cmd(netsh_set_cmd %
+                                           (ifnames[num], "disable"))
+                time.sleep(random.randint(1, 5))
+                index += 1
```

Comment:
Still this part, 
disable 4 nics the same time is invalid

Reply:
I have solved this problem.